node/node.go

```diff
@@ -255,6 +260,17 @@ func (node *Node) reducePendingTransactions() {
 	}
 }
 
+func (node *Node) reducePendingStakingTransactions() {
+	txPoolLimit := core.ShardingSchedule.MaxTxPoolSizeLimit()
+	curLen := len(node.pendingStakingTransactions)
+
+	// If length of pendingStakingTransactions is greater than TxPoolLimit then by greedy take the TxPoolLimit recent transactions.
+	if curLen > txPoolLimit+txPoolLimit {
+		node.pendingStakingTransactions = append(types2.StakingTransactions(nil), node.pendingStakingTransactions[curLen-txPoolLimit:]...)
```

Comment:
we can first detect duplicated staking transactions, then remove the older transactions

Reply:
e.g.  remove the duplicated staking in add pending transactions function to save memory

Reply:
Done.